packages/template-retail-react-app/app/components/list-menu/index.jsx

```diff
@@ -217,40 +220,52 @@ ListMenuPopover.propTypes = {
  * The submenus are open when the user moves the mouse over the trigger and for A11y when
  * users use the keyboard Tab key to focus over the chevron icon and press Enter.
  *
- * @param root The root category in your commerce cloud back-end.
  * @param maxColumns The maximum number of columns that we want to use per row inside the ListMenu.
  */
-const ListMenu = ({root, maxColumns = MAXIMUM_NUMBER_COLUMNS}) => {
+const ListMenu = ({maxColumns = MAXIMUM_NUMBER_COLUMNS}) => {
+    const {root, itemsKey} = useCategories()
     const theme = useTheme()
     const {baseStyle} = theme.components.ListMenu
 
-    const itemsKey = 'categories'
-    const items = root ? root[itemsKey] : false
+    useEffect(() => {
+        var spinner = document.getElementById('spinner')
```

Comment:
I would assume it's a bit tough to guarantee there is exactly one item with `id="spinner"` as a third party library might even use this ID without us realizing. You should add conditional attributes to `<Spinner />` if at all possible. Down here https://github.com/SalesforceCommerceCloud/pwa-kit/pull/875/files#diff-d9e1da2dc459440ee140147b91bec239d587583ec1c5055028687a13b8f3f292R268

Reply:
More generally, is there any way to do this within React, without reaching for native DOM manipulation? Or is this the correct react-y way of doing things?

Reply:
The react-y way is using Refs. See https://github.com/SalesforceCommerceCloud/pwa-kit/pull/875/files#r1063028178